src/components/description/DescriptionSubscriber.js

```diff
@@ -0,0 +1,38 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PubSub from 'pubsub-js';
+import { METADATA_ADD } from '../../events';
+import TitleInfo from '../TitleInfo';
+import Description from './Description';
+
+export default function DescriptionSubscriber(props) {
+  const { description, onReady, removeGridComponent } = props;
+
+  const onReadyCallback = useCallback(onReady, []);
+  const [metadata, setMetadata] = useState({});
+
+  const handleMetadataAdd = useCallback((msg, { name: layerName, metadata: layerMetadata }) => {
```

Comment:
- Probably don't use `layer`? I think this is more general?
- Could be confused, but `metadata` and `layerMetadata` are different kinds of objects? Perhaps `metadata` and `metadataRecord`?

Reply:
True, I like the `metadataRecord` solution, thanks!